server/routes/get_agent.ts

```diff
@@ -26,3 +26,33 @@ export const getAgent = async (id: string, client: OpenSearchClient['transport']
     throw new Error(`get agent ${id} failed, reason: ${errorMessage}`);
   }
 };
+
+export const searchAgentByName = async (name: string, client: OpenSearchClient['transport']) => {
+  try {
+    const requestParams = {
+      query: {
+        term: {
+          'name.keyword': name,
+        },
+      },
+      sort: {
+        created_time: 'desc',
+      },
+      size: 1,
+    };
+
+    const response = await client.request({
+      method: 'GET',
+      path: `${ML_COMMONS_BASE_API}/agents/_search`,
+      body: requestParams,
+    });
+
+    if (!response || response.body.hits.total.value === 0) {
+      throw new Error(`cannot find any agent by name: ${name}`);
+    }
```

Comment:
I think we don't need to throw error when rag agent not been configured?

Reply:
I just imitate the previous implementation of `search agent` in this [PR](https://github.com/opensearch-project/dashboards-assistant/commit/4e065f1ccffbfbbac1a97c942693a88804bba860#diff-c20d6355af792c9cdb22f0f87954d400a8cd0ce0ad3c67efec7bf216947d1eb8R39).

But since we already catch the error where we invoke this function, It's both ok to throw error or return undefined I think.

Reply:
I've changed to return undefined if not finding agent